core/src/test/java/io/methvin/watchservice/DirectoryWatcherOnDiskTest.java

```diff
@@ -108,7 +133,73 @@ public void moveSubDirectory() throws IOException, ExecutionException, Interrupt
         FileUtils.deleteDirectory(newParent.toFile());
       }
     }
+  }
+
+  @Test
+  public void emitCreateEventWhenFileLockedNoHashing() throws IOException, ExecutionException, InterruptedException {
+    this.watcher = DirectoryWatcher.create(Collections.singletonList(this.tmpDir), this.recorder, false);
+    final CompletableFuture future = this.watcher.watchAsync();
+    Random random = new Random();
+    int i = random.nextInt(100_000);
+    final Path child = tmpDir.resolve("child-" + i + ".dat");
+    FileChannel channel = null;
+    FileLock lock = null;
+    try {
 
+      File file = child.toFile();
+      channel = new RandomAccessFile(file, "rw").getChannel();
+      lock = channel.lock();
+
+      try {
+        future.get(5, TimeUnit.SECONDS);
+      } catch (TimeoutException e) {
+        // Expected exception.
+      }
+
+      assertEquals("Create event for the child file was notified", 1, this.recorder.events.size());
+
+      assertTrue(
+          "Create event for the child file was notified",
+          this.recorder.events.stream().anyMatch(
+              e -> e.eventType() == DirectoryChangeEvent.EventType.CREATE && e.path().getFileName().equals(child.getFileName())));
+
+    } finally {
+      if (lock != null && channel != null && channel.isOpen()) {
+        lock.release();
+        channel.close();
+      }
+    }
+  }
+
+  @Test
+  public void doNotEmitCreateEventWhenFileLockedWithHashing() throws IOException, ExecutionException, InterruptedException {
+    this.watcher = DirectoryWatcher.create(Collections.singletonList(this.tmpDir), this.recorder, true);
+    final CompletableFuture future = this.watcher.watchAsync();
+    Random random = new Random();
+    int i = random.nextInt(100_000);
+    final Path child = tmpDir.resolve("child-" + i + ".dat");
+    FileChannel channel = null;
+    FileLock lock = null;
+    try {
+
+      File file = child.toFile();
+      channel = new RandomAccessFile(file, "rw").getChannel();
+      lock = channel.lock();
+
+      try {
+        future.get(5, TimeUnit.SECONDS);
+      } catch (TimeoutException e) {
+        // Expected exception.
+      }
+
+      assertEquals("No event for the file creation expected", 0, this.recorder.events.size());
```

Comment:
Yes, file locking is platform-dependent and it might be difficult to reproduce in all platforms. We can enable it only in windows, where we see it's working. That way we'll have  a demonstration of a file hashing problem that this PR is trying to resolve. What do you think?